client/src/app/tabs/form/FormEditor.js

```diff
@@ -253,22 +264,20 @@ export class FormEditor extends CachedComponent {
 
     this.setState(newState);
 
-    const engineProfile = this.getEngineProfile();
+    try {
+      const engineProfile = this.engineProfile.get();
 
-    const { engineProfile: cachedEngineProfile } = this.getCached();
+      this.engineProfile.setCached(engineProfile);
+    } catch (err) {
 
-    if (!engineProfilesEqual(engineProfile, cachedEngineProfile) && isKnownEngineProfile(engineProfile)) {
-      this.setCached({
-        engineProfile
-      });
+      // TODO
```

Comment:
TODO: Fill with life. When could this throw?

Reply:
I don't really like this at the moment. We only want to update the cached engine profile if we've got a known profile. There isn't a real error case. We just ignore unknown engine profiles.

Reply:
Shall we unset the cached version to `null` in the error case?